### ISSUE TITLE
relaxRequired should not traverse across "not" relationships

### ISSUE THREAD

**Issue by smikulcik** [stoplightio/spectral#3068]
**Describe the bug**

https://github.com/stoplightio/spectral/blob/cee53a30948b95e2bc1df4e1e8615f73e70ad997/packages/rulesets/src/oas/functions/oasExample.ts#L194

Sometimes we use a "not" schema as a workaround for when it's infeasible to add "additionalProperties: false"/"unevaluatedProperties: false". This check is triggering when we use a "not" schema in request schemas.

I think the fix is to adjust the relaxRequired to halt traversal across "not" schemas.

**To Reproduce**

```yaml
openapi: 3.0.3
info:
  title: Foo API
  version: 0.0.1
paths:
  /foos:
    post:
      operationId: create_foo
      summary: Create a foo
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/FooPrototype'
            example:
              name: my-foo
      responses:
        '201':
          description: Created
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Foo'

components:
  schemas:
    FooID:
      type: object
      required: [id]
      properties:
        id:
          type: string
          readOnly: true   # ← BUG: relaxRequired strips id from required[],
                           #   making `not` match any object → example fails
    FooPrototype:
      type: object
      not:
        $ref: '#/components/schemas/FooID'
      required: [name]
      properties:
        name:
          type: string
    Foo:
      type: object
      required: [id, name]
      properties:
        id:
          type: string
          readOnly: true   # ✅ safe — response schema only
        name:
          type: string

```

```sh
$ npx spectral lint ./example.openapi.yaml

example.openapi.yaml
...
 16:21  error  oas3-valid-media-example      "example" property must not be valid                                                                       paths./foos.post.requestBody.content.application/json.example
...
```

**Expected behavior**

We should not see `error  oas3-valid-media-example`

**Screenshots**

N/A

**Environment (remove any that are not applicable):**
 - Library version: 6.11.1
 - OS: Apple M1 Max Tahoe 26.6.2
 - Browser: N/A

**Additional context**
